packages/relay/src/relayer.ts

```diff
@@ -136,13 +164,31 @@ function isApiCredentials(credentials: AutotaskRelayerParams | ApiRelayerParams)
   return !!apiCredentials.apiKey && !!apiCredentials.apiSecret;
 }
 
+// If a tx-like object is representing a legacy transaction (type 0)
+export function isLegacyTx<TransactionLikeType>(
+  tx: TransactionLikeType,
+): tx is TransactionLikeType & { gasPrice: unknown } {
+  return 'gasPrice' in tx;
```

Comment:
one thing I would mention is that it could be possible that this prop is somehow included in an object (but `undefined`) and this would return `true`:

```javascript
const obj = { a: undefined }
// undefined
!!obj.a
// false
'a' in obj
// true
```
It's slightly nitpicky, but unless there's a reason why the mere existence of the property is relevant (rather than it having value), I like to just check the truthiness (of course you can run into issues with `0` in this case...)

Reply:
What a nice catch @mok0230!

This is super tricky, but it's an agreement we made (Palla, Fran, and me) in the backend since there are around 5 different types of `Transaction` types representing different things and having multiple `gasPrice`, `maxFeePerGas` and `maxPriorityFeePerGas`. Turns out that in both the backend and here there's no way to have null or undefined since types are mutually exclusive (between legacy and EIP1559) by design, but this is a great enhancement for both client and backend.

I added this on both.